File: 4248-count-commas-in-range-ii/count-commas-in-range-ii.py
class Solution:
    def countCommas(self, n: int) -> int:
        res = 0

        lower = 1000

        commas = 1

        while lower <= n:
            upper = lower*1000-1
            if upper > n : upper = n
            countcommas = upper-lower +1
            res += (countcommas*commas)

            lower *= 1000
            commas += 1
        
        return res

File: 4248-count-commas-in-range-ii/count-commas-in-range-ii.java
class Solution {
    public long countCommas(long n) {
        long res = 0;

        long lower = 1000;

        long commas = 1;

        while (lower<=n){
            long upper = lower*1000-1;
            if (upper > n) upper = n;

            long countcommas = upper-lower+1;
            
            res += countcommas*commas;

            lower *= 1000;
            commas ++;

        }

        return res;

    }
}